docs/templates/plugin.rst.j2

```diff
@@ -269,8 +283,20 @@ Facts returned by this module are added/updated in the ``hostvars`` host facts a
                     <td class="elbow-placeholder"></td>
                 {% endfor %}
                 <td colspan="@{ from_kludge_ns('maxdepth') - loop.depth0 }@" colspan="@{ from_kludge_ns('maxdepth') - loop.depth0 }@">
-                    <b>@{ key }@</b>
-                    <br/><div style="font-size: small; color: red">@{ value.type }@</div>
+                    <b>@{ key }@</b><br/>
+                    {# Clean up parameter types #}
+                    {% if value.type is not defined or value.type == 'str' %}
+                        {% set type = 'string' %}
+                    {% elif value.type == 'bool' %}
+                        {% set type = 'boolean' %}
+                    {% elif value.type == 'int' %}
+                        {% set type = 'integer' %}
+                    {% elif value.type == 'dict' %}
+                        {% set type = 'dictionary' %}
+                    {% else %}
+                        {% set type = value.type %}
```

Comment:
The idea is that in the case of `raw`, we can still override the type in the documentation. In most cases we have a preference in these cases, and we use `raw` either because of backward compatibility (we have to support e.g. both list and dicts) or because of exceptional circumstances.

Reply:
If that's the case, I would add handling of `raw` to print out something more useful for the user. The string `raw` itself is less useful (and more confusing) for most users than not showing the type (as we are currently doing) in my opinion.

Reply:
I agree that this discussion should be moved out of this PR, but I'd really like something more helpful than `raw` to be printed out by this PR :)

Reply:
I disagree that `raw` is less useful than not showing a type. At least you have something to work with, and look up/ask around what it means. It gives a clue, even if not the answer you were expecting.

Again, in most (if not all) cases we want the author to override the type in the documentation, but in some cases `raw` might be the best thing to return.

We intend to do a cleanup of certain things, and documenting the `raw` type parameters would be one. (In some cases the parameter description would say in this case: *This parameter accepts `list` or `dict` input, look at the examples on how this works.* Similar to how we now handle conditional defaults.